components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3194,11 +2903,662 @@ subroutine rain_immersion_freezing(t,    &
+   ! these values to map back to cell-average quantities where applicable.
+
+          ! map warm-phase process rates to cell-avg
+   qcacc   = qcacc*lr_cldm     ! Accretion of liquid to rain
+   qrevp   = qrevp*rcldm       ! Evaporation of rain
+   qcaut   = qcaut*lcldm       ! Autoconversion of liquid
+   ncacc   = ncacc*lr_cldm     ! Number change due to accretion
+   ncslf   = ncslf*lcldm       ! Self collection occurs locally in liq. cloud
+   ncautc  = ncautc*lcldm      ! Impact of autoconversion on number
+   nrslf   = nrslf*rcldm       ! Self collection occurs locally in rain cloud
+   nrevp   = nrevp*rcldm       ! Change in rain number due to evaporation 
+   ncautr  = ncautr*lr_cldm    ! Autoconversion of rain drops within rain/liq cloud
+     ! AaronDonahue: These variables are related to aerosol activation and their usage will be changed in a later PR.
+   qcnuc   = qcnuc*lcldm       ! Impact on liq. from nucleation
+   ncnuc   = ncnuc*lcldm       ! Number change due to aerosol activation
+
+   ! map ice-phase  process rates to cell-avg
+   qisub   = qisub*icldm       ! Sublimation of ice in ice cloud
+   nrshdr  = nrshdr*il_cldm    ! Rain # increase due to shedding from rain-ice collisions, occurs when ice and liquid interact
+   qcheti  = qcheti*il_cldm    ! Immersion freezing of cloud drops
+   qrcol   = qrcol*ir_cldm     ! Collection of rain mass by ice
+   qcshd   = qcshd*il_cldm     ! Rain mass growth due to shedding of fain drops after collisions with ice, occurs when ice and liquid interact
+   qimlt   = qimlt*icldm       ! Melting of ice
+   qccol   = qccol*il_cldm     ! Collection of water by ice
+   qrheti  = qrheti*rcldm      ! Immersion freezing of rain
+   nimlt   = nimlt*icldm       ! Change in number due to melting
+   nccol   = nccol*il_cldm     ! Cloud # change due to collection of cld water by ice
+   ncshdc  = ncshdc*il_cldm    ! Number change due to shedding, occurs when ice and liquid interact
+   ncheti  = ncheti*lcldm      ! Number change associated with freexzing of cld drops
+   nrcol   = nrcol*ir_cldm     ! Rain number change due to collection from ice
+   nislf   = nislf*icldm       ! Ice self collection
+   qidep   = qidep*icldm       ! Vapor deposition to ice phase
+   nrheti  = nrheti*rcldm      ! Change in number due to immersion freezing of rain
+   nisub   = nisub*icldm       ! Number change due to sublimation of ice
+   qiberg  = qiberg*il_cldm    ! Bergeron process
+     ! AaronDonahue: These variables are related to aerosol activation and their usage will be changed in a later PR.
+   qinuc   = qinuc             ! Deposition and condensation-freezing nucleation, already cell-averaged
+   ninuc   = ninuc             ! Number change due to deposition and condensation-freezing, already cell-averaged
+
+end subroutine back_to_cell_average 
+
+subroutine prevent_ice_overdepletion(pres,t,qv,xxls,odt,    &
```

Comment:
Thank you for making this it's own process.  It will make it easier for us to reassess how we want to handle this in the future.